waspc/src/Wasp/Psl/Parser/Enum.hs

```diff
@@ -0,0 +1,47 @@
+module Wasp.Psl.Parser.Enum
+  ( enum,
+  )
+where
+
+import Text.Parsec
+  ( many,
+    many1,
+    try,
+    (<|>),
+  )
+import Text.Parsec.String (Parser)
+import qualified Wasp.Psl.Ast.Schema as Psl.Ast
+import Wasp.Psl.Parser.Common
+  ( blockAttribute,
+    braces,
+    fieldAttribute,
+    identifier,
+    reserved,
+    whiteSpace,
+  )
+
+-- | Parses PSL (Prisma Schema Language enum).
+-- Example of PSL enum:
+--   enum Role {
+--     USER @map("user")
+--     ADMIN
+--     @@map("role")
+--   }
+enum :: Parser Psl.Ast.SchemaElement
```

Comment:
I think I need to keep this in order to make the `schema` parser work:
```hs
schema :: Parser Model.Schema
schema = do
  whiteSpace
  elements <-
    many
      ( try model
          <|> try enum
          <|> try configBlock
      )
  return $ Model.Schema elements
```

Reply:
I might have made some mistake here, but the idea should be clear:

```hs
schema :: Parser Model.Schema
schema = do
  whiteSpace
  elements <- many $ choice
    [ ModelBlock <$> try model
    , ModelEnum <$> try enum
    , ModelConfigBlock <$> try configBlock
    ]
  return $ Model.Schema elements
```

Notice I used `choice` for a bit more readability.

The important part is I apply data constructors here. And now your model parser can return Model that is not wrapped. That is better separation of concerns (is that correct term?) than what you have now.

Reply:
I've adjusted the code to use `choice` instead of `<|>` in a few places and I've also taken out the `Schema.*Block` wrappers out the respective parsers 👍